File: src/main/scala/keri/projectx/block/BlockGlassViewer.java
/*
 * Copyright (c) 2017 KitsuneAlex & Adam8234. All rights reserved!
 * Do not distribute or redistribute without the explicit permission
 * of the developer!
 */

package keri.projectx.block;

import keri.ninetaillib.lib.render.connected.ICTMBlock;
import keri.ninetaillib.lib.texture.IIconRegister;
import keri.ninetaillib.lib.util.ClientUtils;
import keri.projectx.util.ModPrefs;
import net.minecraft.block.material.Material;
import net.minecraft.block.state.IBlockState;
import net.minecraft.client.renderer.texture.TextureAtlasSprite;
import net.minecraft.util.BlockRenderLayer;
import net.minecraft.util.EnumFacing;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IBlockAccess;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public class BlockGlassViewer extends BlockProjectX implements ICTMBlock {

    @SideOnly(Side.CLIENT)
    private TextureAtlasSprite[] texture;

    public BlockGlassViewer() {
        super("glass_viewer", Material.GLASS);
        this.setHardness(1.4F);
    }

    @Override
    @SideOnly(Side.CLIENT)
    public void registerIcons(IIconRegister register) {
        this.texture = ClientUtils.registerConnectedTexture(register, ModPrefs.MODID + ":blocks/glass_viewer/glass_viewer");
    }

    @Override
    @SideOnly(Side.CLIENT)
    public TextureAtlasSprite getIcon(int meta, EnumFacing side) {
        return this.texture[0];
    }

    @Override
    @SideOnly(Side.CLIENT)
    public boolean canTextureConnect(IBlockAccess world, BlockPos pos, EnumFacing side) {
        return true;
    }

    @Override
    @SideOnly(Side.CLIENT)
    public TextureAtlasSprite[] getConnectedTexture(IBlockAccess world, BlockPos pos, EnumFacing side) {
        return this.texture;
    }

    @Override
    @SideOnly(Side.CLIENT)
    @SuppressWarnings("deprecation")
    public boolean isOpaqueCube(IBlockState state) {
        return false;
    }

    @Override
    @SideOnly(Side.CLIENT)
    @SuppressWarnings("deprecation")
    public boolean isFullCube(IBlockState state) {
        return false;
    }

    @Override
    @SideOnly(Side.CLIENT)
    public boolean canRenderInLayer(IBlockState state, BlockRenderLayer layer) {
        return layer == BlockRenderLayer.TRANSLUCENT;
    }

    @Override
    @SideOnly(Side.CLIENT)
    @SuppressWarnings("deprecation")
    public boolean shouldSideBeRendered(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side) {
        return state.getBlock() != this;
    }

}

File: src/main/scala/keri/projectx/block/BlockGlassViewer.scala
package keri.projectx.block

import keri.ninetaillib.lib.render.connected.ICTMBlock
import keri.ninetaillib.lib.texture.IIconRegister
import keri.ninetaillib.lib.util.ClientUtils
import keri.projectx.util.ModPrefs
import net.minecraft.block.material.Material
import net.minecraft.block.state.IBlockState
import net.minecraft.client.renderer.texture.TextureAtlasSprite
import net.minecraft.util.math.BlockPos
import net.minecraft.util.{BlockRenderLayer, EnumFacing}
import net.minecraft.world.IBlockAccess
import net.minecraftforge.fml.relauncher.{Side, SideOnly}

class BlockGlassViewer extends BlockProjectX("glass_viewer", Material.GLASS) with ICTMBlock {
  setHardness(1.1F)

  @SideOnly(Side.CLIENT)
  var texture: Array[TextureAtlasSprite] = _

  @SideOnly(Side.CLIENT)
  override def registerIcons(register: IIconRegister): Unit = {
    texture = ClientUtils.registerConnectedTexture(register, ModPrefs.MODID + ":blocks/glass_viewer/glass_viewer")
  }

  @SideOnly(Side.CLIENT)
  override def getIcon(meta: Int, side: EnumFacing): TextureAtlasSprite = texture(0)

  @SideOnly(Side.CLIENT)
  override def getConnectedTexture(world: IBlockAccess, pos: BlockPos, side: EnumFacing): Array[TextureAtlasSprite] = texture

  @SideOnly(Side.CLIENT)
  override def canTextureConnect(world: IBlockAccess, pos: BlockPos, side: EnumFacing): Boolean = true

  @SideOnly(Side.CLIENT)
  override def shouldSideBeRendered(state: IBlockState, world: IBlockAccess, pos: BlockPos, side: EnumFacing): Boolean = {
    world.getBlockState(pos.offset(side)).getBlock != this
  }

  @SideOnly(Side.CLIENT)
  override def canRenderInLayer(state: IBlockState, layer: BlockRenderLayer): Boolean = layer == BlockRenderLayer.TRANSLUCENT

  @SideOnly(Side.CLIENT)
  override def isOpaqueCube(state: IBlockState): Boolean = false

  @SideOnly(Side.CLIENT)
  override def isFullCube(state: IBlockState): Boolean = false

}
